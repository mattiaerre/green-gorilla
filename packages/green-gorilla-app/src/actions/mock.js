const mock = {
  info: {
    name: 'red-rhinoceros',
    version: '2.0.0-alpha.4'
  }
};

export default mock;
