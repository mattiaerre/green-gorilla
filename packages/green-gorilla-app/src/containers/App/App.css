.App {
  text-align: center;
}

.OpenWeatherMap {
  padding: 1em;
}

.buttons-container {
  padding: 1em;
}

.awesome {
  background-color: #5cb85c;
}

.good {
  background-color: #5bc0de;
}

.not-great {
  background-color: #f0ad4e;
}

.unhappy {
  background-color: #d9534f;
}

.buttons-container button {
  border: 1px solid gray;
  background-color: lightgray;
  margin: 0.25em;
  padding: 0.5em;
}
