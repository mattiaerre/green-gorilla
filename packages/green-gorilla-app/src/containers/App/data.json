[
  {
    "city": "San Francisco"
  },
  {
    "city": "London"
  },
  {
    "city": "Torino"
  },
  {
    "city": "Tokyo"
  }
]
