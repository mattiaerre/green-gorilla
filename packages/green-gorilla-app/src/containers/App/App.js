import React, { Component } from 'react';
import OpenWeatherMap from 'react-open-weather-map';
import Header from '../../components/Header/Header';
import data from './data';
import MoraleCheck from '../../components/MoraleCheck/MoraleCheck';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      info: { name: '', version: '' },
      morale: props.morale ? props.morale : 'good'
    };

    this.getData = this.getData.bind(this);
    this.handleOnClick = this.handleOnClick.bind(this);
  }

  getData(city) {
    this.props.actions.getData(city).then(data => {
      this.setState(data);
    });
  }

  componentDidMount() {
    this.getData(data[2].city);
  }

  handleOnClick(e) {
    this.getData(e.target.value);
  }

  render() {
    const buttons = data.map(({ city }) => (
      <button type="button" key={city} value={city} onClick={this.handleOnClick}>{city}</button>)
    );
    return (
      <div className={`App ${this.state.morale}`}>
        <Header {...this.state.info} />
        <div className="buttons-container">
          {buttons}
        </div>
        <OpenWeatherMap
          config={{ containerClassName: 'OpenWeatherMap' }}
          data={this.state.currentWeather} />
        <MoraleCheck onClick={(id) => { this.setState({ morale: id }); }} />
      </div>
    );
  }
}

export default App;
