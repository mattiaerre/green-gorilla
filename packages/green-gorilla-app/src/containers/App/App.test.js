import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';
import mock from '../../actions/mock';

describe('<App />', () => {
  const getData = Promise.resolve(mock);

  const props = {
    actions: {
      getData: () => (getData)
    }
  }

  test('constructor', () => {
    const component = create(
      <App {...props} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('componentDidMount', async () => {
    const component = create(
      <App {...props} />
    );

    await getData;

    expect(component.toJSON()).toMatchSnapshot();
  });
});
