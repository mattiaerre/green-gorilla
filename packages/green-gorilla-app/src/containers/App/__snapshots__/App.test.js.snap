// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> componentDidMount 1`] = `
<div
  className="App good"
>
  <div
    className="Header"
  >
    <h1>
      red-rhinoceros
    </h1>
    <h2>
      2.0.0-alpha.4
    </h2>
  </div>
  <div
    className="buttons-container"
  >
    <button
      onClick={[Function]}
      type="button"
      value="San Francisco"
    >
      San Francisco
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="London"
    >
      London
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="Torino"
    >
      Torino
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="Tokyo"
    >
      Tokyo
    </button>
  </div>
  <div
    className="OpenWeatherMap"
  >
    <div
      className="main-heading"
    >
      London
      , 
      GB
    </div>
    <div
      className="day"
    >
      Monday 6:50 AM
    </div>
    <div
      className="description"
    >
      shower rain
    </div>
    <div
      className="icon"
    >
      <img
        alt="shower rain"
        src="http://openweathermap.org/img/w/09d.png"
      />
    </div>
    <div
      className="temperature"
    >
      23.33
       °C
    </div>
  </div>
  <ul
    className="morale-check"
  >
    <li
      className="awesome"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        😎
      </span>
      <span
        className="description"
      >
        I'm feeling awesome
      </span>
    </li>
    <li
      className="good"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        👍
      </span>
      <span
        className="description"
      >
        I'm feeling good
      </span>
    </li>
    <li
      className="not-great"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        😕
      </span>
      <span
        className="description"
      >
        I'm not feeling great
      </span>
    </li>
    <li
      className="unhappy"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        😩
      </span>
      <span
        className="description"
      >
        I'm feeling unhappy
      </span>
    </li>
  </ul>
</div>
`;

exports[`<App /> constructor 1`] = `
<div
  className="App good"
>
  <div
    className="Header"
  >
    <h1>
      
    </h1>
    <h2>
      
    </h2>
  </div>
  <div
    className="buttons-container"
  >
    <button
      onClick={[Function]}
      type="button"
      value="San Francisco"
    >
      San Francisco
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="London"
    >
      London
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="Torino"
    >
      Torino
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="Tokyo"
    >
      Tokyo
    </button>
  </div>
  <div
    className="OpenWeatherMap"
  />
  <ul
    className="morale-check"
  >
    <li
      className="awesome"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        😎
      </span>
      <span
        className="description"
      >
        I'm feeling awesome
      </span>
    </li>
    <li
      className="good"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        👍
      </span>
      <span
        className="description"
      >
        I'm feeling good
      </span>
    </li>
    <li
      className="not-great"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        😕
      </span>
      <span
        className="description"
      >
        I'm not feeling great
      </span>
    </li>
    <li
      className="unhappy"
      onClick={[Function]}
    >
      <span
        className="emoji"
      >
        😩
      </span>
      <span
        className="description"
      >
        I'm feeling unhappy
      </span>
    </li>
  </ul>
</div>
`;
